### PR TITLE
Fixing test_build_sets_from_multiple_threads/test.py::test_set

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit ef796c668d187c80f13223dbf679dcc442907008 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61286/ef796c668d187c80f13223dbf679dcc442907008/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Planner/Planner.cpp [hunk 1]
```diff
@@ -1081,7 +1081,9 @@ void addBuildSubqueriesForSetsStepIfNeeded(
     for (auto & subquery : subqueries)
     {
         auto query_tree = subquery->detachQueryTree();
-        auto subquery_options = select_query_options.subquery();
+        /// I suppose it should be better to use all flags from select_query_options,
+        /// But for now it is done in the same way as in old analyzer.
```
Comment (novikd) [inline src/Planner/Planner.cpp:1085]: What was the difference? It can affect the performance because it can for example reset `only_analyze`, can't it?

Comment (KochetovNicolai) [reply src/Planner/Planner.cpp:1085]: only_analyze should be fine, cause subqueries won't be actually executed

Comment (alexey-milovidov) [reply src/Planner/Planner.cpp:1086]: Could you please also clarify, what it affects?
(you can write directly in the code)

**Pull request merged**